imap_data_access/file_validation.py

```diff
@@ -356,3 +356,259 @@ def construct_path(self) -> Path:
         # Use the file suffix to determine the directory structure
         # IMAP_DATA_DIR/imap/spice/<subdir>/filename
         return spice_dir / subdir / self.filename
+
+
+class AncillaryFilePath:
+    """Class for building and validating filepaths for Ancillary files."""
+
+    class InvalidAncillaryFileError(Exception):
+        """Indicates a bad file type."""
+
+        pass
+
+    def __init__(self, filename: str | Path):
+        """Class to store filepath and file management methods for Ancillary files.
+
+        If you have an instance of this class, you can be confident you have a valid
+        ancillary file and generate paths in the correct format. The parent of the file
+        path is set by the "IMAP_DATA_DIR" environment variable, or defaults to "data/"
+
+        Current filename convention:
+        "<mission>_<instrument>_<ancillary_name>_<start_date>(-<end_date>)_
+        <version>.<extension>"
+
+        <mission>: imap
+        <instrument>: codice, glows, hi, hit, idex, lo, mag, swapi, swe, ultra
+        <ancillary_name>: A descriptive name for the ancillary file which
+        distinguishes between other ancillary files used by the instrument.
+        <start_date>: startdate is the earliest date in the data, format: YYYYMMDD
+        <end_date>: The end time of the validity of the ancillary file,
+        in the format “YYYYMMDD”. This is optional for files, with the understanding
+         that if end_date is not provided, the file is valid until a file with a
+         later start_date and no end_date.
+        <version>: This stores the data version for this product, format: vXXX
+
+        Parameters
+        ----------
+        filename : str | Path
+            Ancillary data filename or file path.
+        """
+        self.filename = Path(filename)
+        self.data_dir = imap_data_access.config["DATA_DIR"]
+
+        try:
+            split_filename = self.extract_filename_components(self.filename)
+        except ValueError as err:
+            raise self.InvalidAncillaryFileError(
+                f"Invalid filename. Expected file to match format: "
+                f"{imap_data_access.ANCILLARY_FILENAME_CONVENTION}"
+            ) from err
+
+        self.mission = split_filename["mission"]
+        self.instrument = split_filename["instrument"]
+        self.ancillary_name = split_filename["ancillary_name"]
```

Comment:
If we make this 'descriptor" then we can actually match all the attributes of ScienceFilePath and share a lot more code. Plus, this piece does still act as a descriptor.